components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -888,14 +882,8 @@ func checkIfRoleIntersectsProjectsFilter(ctx context.Context, q Querier,
 
 	// Return true or false if there is intersection between iam_role_projects and projectsFilter,
 	// assuming '{(unassigned)}' in the case that iam_role_projects is empty. If a role of id
-	// doesn't exist, this will return 0 rows which will bubble up to NotFoundErr when passed to processError.
-	row := q.QueryRowContext(ctx,
-		`SELECT COALESCE(array_agg(rp.project_id)
-				FILTER (WHERE rp.project_id IS NOT NULL), '{(unassigned)}') && $2 AS intersection
-			FROM iam_roles AS r
-			LEFT OUTER JOIN iam_role_projects AS rp ON rp.role_id=r.db_id
-			WHERE r.id = $1;`,
-		id, pq.Array(projectsFilter))
+	// doesn't exist, this will return a proper SQL "no rows" error when passed to processError.
+	row := q.QueryRowContext(ctx, "SELECT projects_match(role_projects($1), $2)", id, pq.Array(projectsFilter))
```

Comment:
🤔 would make sense to coalesce this into `{(unassigned)}`, but the tests pass regardless...